Std/Data/BitVec/Bitblast.lean

```diff
@@ -0,0 +1,180 @@
+/-
+Copyright (c) 2023 by the authors listed in the file AUTHORS and their
+institutional affiliations. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Harun Khan, Abdalrhman M Mohamed, Joe Hendrix
+-/
+import Std.Data.BitVec.Folds
+
+/-!
+# Bitblasting of bitvectors
+
+We prove the equivalence of a bitblasting operations and their `BitVec` implementation.
+
+## Main results
+* `BV_add`: Bitblasted addition `bitadd` is equivalent to `BitVec.add`.
+
+## Future work
+All other operations are to be PR'ed later and are already proved in
+https://github.com/mhk119/lean-smt/blob/bitvec/Smt/Data/BitVec.lean.
+-/
+
+namespace Std
+
+open Nat Bool
+
+/-! ### Preliminaries -/
+
+namespace BitVec
+
+private
+theorem testBit_limit {x i : Nat} (x_lt_succ : x < 2^(i+1)) : testBit x i = decide (x ≥ 2^i) := by
+  cases xi : testBit x i with
+  | true =>
+    simp [testBit_implies_ge xi]
+  | false =>
+    simp
+    cases Nat.lt_or_ge x (2^i) with
+    | inl x_lt =>
+      exact x_lt
+    | inr x_ge =>
+      have ⟨j, ⟨j_ge, jp⟩⟩  := ge_two_pow_implies_high_bit_true x_ge
```

Comment:
`rcases` is nice:
```
    rcases Nat.lt_or_ge x (2^i) with x_lt | x_ge 
    · exact x_lt
    · have ⟨j, ⟨j_ge, jp⟩⟩  := ge_two_pow_implies_high_bit_true x_ge
```
Here it is only saving a few keystrokes and some vertical space, but it is a lovely tactic (you can destruct using patterns) that we shouldn't hesitate to use.